docs/dev/architeture.md

```diff
@@ -0,0 +1,208 @@
+
+### Pluggable Table Engine
+module path: https://github.com/CeresDB/ceresdb/tree/main/table_engine
+
+`Table Engine` is actually a storage engine for managing tables in CeresDB and the pluggability of `Table Engine` is a core design of CeresDB which matters in achieving our target (process both timeseries and analytic workloads well). CeresDB will have multiple kinds of `Table Engine` for different workloads and the most appropriate one should be chosen as the storage engine according to the workload pattern.
+
+Now the requirements for a `Table Engine` are:
+- Manage all the shared resources under the engine:
+    - Memory
+    - Storage
+    - CPU
+- Manage metadata of tables such as table schema and table options;
+- Provide `Table` instances which provides `read` and `write` methods;
+- Take responsibilities for creating, opening, dropping and closing `Table` instance;
+- ....
+
+Actually the things that a `Table Engine` needs to process are a little complicated. And now in CeresDB only one `Table Engine` called `Analytic` is provided and does a good job in processing analytical workload, but it is not ready yet to handle the timeseries workload (we plan to enhance it for a better performance by increasing some indexes which helps handle timeseries workload).
+
+The following part gives a description about details of `Analytic Table Engine`.
+
+#### WAL
+module path: https://github.com/CeresDB/ceresdb/tree/main/wal
+
+The model of CeresDB processing data is `WAL` + `MemTable` that the recent written data is written to `WAL` first and then to `MemTable` and after a certain amount of data in `MemTable` is accumulated, the data will be organized in a query-friendly form to persistent devices.
+
+Now two implementations of `WAL` are provided for stand-alone and distributed mode:
+- For stand-alone mode, `WAL` is based on `RocksDB` and data is persisted on the local disk.
+- For distributed mode, `WAL` is required as a distributed component and to be responsible for reliability of the newly written data, so now we provide an implementation based on [OceanBase](https://github.com/oceanbase/oceanbase) and in our roadmap a more lightweight implementation will be provided.
+
+Besides, `WAL`'s trait definition tells that `WAL` has the concept of `Region` and actually each table is assigned to a `Region` so that the isolation between tables is gained and such an isolation provides convenience for some operations on table's level (such as different `TTL`s for different tables).
+
+#### MemTable
+module path: https://github.com/CeresDB/ceresdb/tree/main/analytic_engine/src/memtable
+
+`Memtable` is used to store the newly written data and after a certain amount of data is accumulated, CeresDB organizes the data in `MemTable` into a query-friendly storage format (`SST`) and stores it to the persistent device. Before persisting, `MemTable` also provides querying.
```

Comment:
```suggestion
`Memtable` is used to store the newly written data and after a certain amount of data is accumulated, CeresDB organizes the data in `MemTable` into a query-friendly storage format (`SST`) and stores it in the persistent device. `MemTable` is readable before it got persisted (flushed).
```

Reply:
- Maybe `stores it to` is better?
- Mabye `before it gets persisted` is better?